contracts/Content/UniqueContent.sol

```diff
@@ -0,0 +1,191 @@
+
+        // increment uniqueIdsCounter for next minting process
+        uniqueIdsCounter++;
+    }
+
+    /** Asset Burning
+    * @dev If the caller is the owner of the token and if the unique asset is not creator locked (or the caller is the creator), it burns the unique asset, returns the original asset, and then deletes the unique asset's token info
+    * @param _uniqueId uint256 ID of token to burn
+    */
+    function burn(uint256 _uniqueId) external override {
+        require(ownerOf(_uniqueId) == _msgSender(), "You must be the owner of this token to burn it");
+        require(uniqueAssetInfo[_uniqueId].creator == _msgSender() || !uniqueAssetInfo[_uniqueId].creatorLocked, "Creator has disabled the burning of this token");
+        _burn(_uniqueId);
+        // transfers original asset back to caller
+        IERC1155Upgradeable(uniqueAssetInfo[_uniqueId].contentAddress).safeTransferFrom(address(this), _msgSender(), uniqueAssetInfo[_uniqueId].tokenId, 1, "");
+        
+        uniqueAssetInfo[_uniqueId].creator = address(0);
+        uniqueAssetInfo[_uniqueId].contentAddress = address(0);
+        uniqueAssetInfo[_uniqueId].tokenId = 0;
+        uniqueAssetInfo[_uniqueId].version = 0;
+        uniqueAssetInfo[_uniqueId].creatorLocked = false;
+        delete uniqueAssetInfo[_uniqueId].uniqueAssetUri;
+        _deleteTokenRoyalties(_uniqueId);
+        
+        emit Burn(_msgSender(), _uniqueId);
+    }
+
+    /**
+    * @dev Returns the uri of a specific version of the original asset the unique asset is based on
+    * @param _uniqueId uint256 ID of token to query original asset uri of
+    * @param _version version number of token to query
+    */
+    function originalAssetUri(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        uint256 _tokenId = uniqueAssetInfo[_uniqueId].tokenId;
+        return IContent(uniqueAssetInfo[_uniqueId].contentAddress).uri(_tokenId, _version);
+    }
+
+    /**
+    * @dev Returns the unique asset uri of a specific version
+    * @param _uniqueId uint256 ID of token to query
+    * @param _version version number of token to query
+    */
+    function tokenURI(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        if (_version > uniqueAssetInfo[_uniqueId].version) {
+            _version = uniqueAssetInfo[_uniqueId].version;
+        } 
+        return uniqueAssetInfo[_uniqueId].uniqueAssetUri[_version];
+    }
+
+    /**
+    * @dev Returns the latest version of the unique asset uri
+    * @param _uniqueId uint256 ID of token to query
+    */
+    function tokenURI(uint256 _uniqueId) public view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        uint256 _version = uniqueAssetInfo[_uniqueId].version;
```

Comment:
don't create a local variable here. just move "uniqueAssetInfo[_uniqueId].version into the uniqueAssetUri[]

creating a local variable increases the cost of the contract